src/eddymotion/model/dipy.py

```diff
@@ -0,0 +1,419 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright 2024 The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+"""DIPY-like models (a sandbox to trial them out before upstreaming to DIPY)."""
+
+from __future__ import annotations
+
+import numpy as np
+from dipy.core.gradients import GradientTable
+from dipy.reconst.base import ReconstModel
+from sklearn.gaussian_process import GaussianProcessRegressor
+
+# from dipy.reconst.multi_voxel import multi_voxel_fit
+
+
+def gp_prediction(
+    model_gtab: GradientTable,
+    gtab: GradientTable,
+    model: GaussianProcessRegressor,
+    mask: np.ndarray | None = None,
+) -> np.ndarray:
+    """
+    Predicts one or more DWI orientations given a model.
+
+    This function checks if the model is fitted and then extracts
+    orientations and potentially b-values from the gtab. It predicts the mean
+    and standard deviation of the DWI signal using the model.
+
+    Parameters
+    ----------
+    model: :obj:`~sklearn.gaussian_process.GaussianProcessRegressor`
+        A fitted GaussianProcessRegressor model.
+    gtab: :obj:`~dipy.core.gradients.GradientTable`
+        Gradient table with one or more orientations at which the GP will be evaluated.
+    mask: :obj:`numpy.ndarray`
+        A boolean mask indicating which voxels to use (optional).
+
+    Returns
+    -------
+    :obj:`numpy.ndarray`
+        A 3D or 4D array with the simulated gradient(s).
+
+    """
+
+    # Check it's fitted as they do in sklearn internally
+    # https://github.com/scikit-learn/scikit-learn/blob/972e17fe1aa12d481b120ad4a3dc076bae736931/\
+    # sklearn/gaussian_process/_gpr.py#L410C9-L410C42
+    if not hasattr(model, "X_train_"):
+        raise RuntimeError("Model is not yet fitted.")
+
+    # Extract orientations from gtab, and highly likely, the b-value too.
+    return model.predict(gtab, return_std=False)
```

Comment:
I am trying to debug this  
https://github.com/nipreps/eddymotion/pull/228#discussion_r1776148824

While at it, I see that `model_gtab` is not used by this method. Although I may imagine what was the underlying intention (much what we try to do in https://github.com/nipreps/eddymotion/pull/228#discussion_r1776148824), not sure how you intended to implement this. This is called from:
https://github.com/nipreps/eddymotion/pull/188/files#diff-fba3c6e90b496595a3cd3cdcbd5d562082cde0b4f21629d667d1a2260a91bd37R281

The mask is not yet used, and although we have discussed using the mask in the past, it should probably be related to the use of `gtab` and `model_gtab`.

@oesteban any thoughts about this?